pkg/grpc/client.go

```diff
@@ -36,36 +35,29 @@ type Client struct {
 	host             string
 	timeoutSeconds   int
 	insecure         bool
-	grpcConnectOnce  *sync.Once
```

Comment:
This shared state causes issues with goroutines.
Even before it wasn't really working. It's just that the concurrency was broken. When that was fixed this started failing :)